### ISSUE TITLE
[BUG]: When working with large datasets, Deepface recognizes irrelevant individuals with a high similarity score.

### ISSUE THREAD

**Issue by rezktr** [serengil/deepface#1523]
### Before You Report a Bug, Please Confirm You Have Done The Following...

- [x] I have updated to the latest version of the packages.
- [x] I have searched for both [existing issues](https://github.com/serengil/deepface/issues) and [closed issues](https://github.com/serengil/deepface/issues?q=is%3Aissue+is%3Aclosed) and found none that matched my issue.

### DeepFace's version

0.0.95

### Python version

3.10

### Dependencies

absl-py==2.3.1
annotated-types==0.7.0
anyio==4.10.0
astunparse==1.6.3
beautifulsoup4==4.13.4
blinker==1.9.0
boto3==1.40.23
botocore==1.40.23
certifi==2025.8.3
charset-normalizer==3.4.3
click==8.2.1
colorama==0.4.6
contourpy==1.3.2
cycler==0.12.1
deepface==0.0.95
exceptiongroup==1.3.0
filelock==3.18.0
fire==0.7.0
Flask==3.1.1
flask-cors==6.0.1
flatbuffers==25.2.10
fonttools==4.59.2
fsspec==2025.9.0
gast==0.6.0
gdown==5.2.0
google-pasta==0.2.0
grpcio==1.74.0
gunicorn==23.0.0
h11==0.16.0
h2==4.2.0
h5py==3.14.0
hpack==4.1.0
httpcore==1.0.9
httpx==0.28.1
hyperframe==6.1.0
idna==3.10
itsdangerous==2.2.0
Jinja2==3.1.6
jmespath==1.0.1
joblib==1.5.1
keras==3.11.2
kiwisolver==1.4.9
libclang==18.1.1
lz4==4.4.4
Markdown==3.8.2
markdown-it-py==4.0.0
MarkupSafe==3.0.2
matplotlib==3.10.6
mdurl==0.1.2
ml_dtypes==0.5.3
mpmath==1.3.0
mtcnn==1.0.0
namex==0.1.0
networkx==3.4.2
numpy==2.1.3
opencv-python==4.12.0.88
opt_einsum==3.4.0
optree==0.17.0
packaging==25.0
pandas==2.3.1
pillow==11.3.0
polars==1.33.1
portalocker==3.2.0
protobuf==5.29.5
psutil==7.0.0
pydantic==2.11.7
pydantic_core==2.33.2
Pygments==2.19.2
pyparsing==3.2.3
PySocks==1.7.1
python-dateutil==2.9.0.post0
pytz==2025.2
pywin32==311
PyYAML==6.0.2
qdrant-client==1.15.1
requests==2.32.4
retina-face==0.0.17
rich==14.1.0
s3transfer==0.13.1
scipy==1.15.3
six==1.17.0
sniffio==1.3.1
soupsieve==2.7
sympy==1.14.0
tensorboard==2.19.0
tensorboard-data-server==0.7.2
tensorflow==2.19.0
tensorflow-io-gcs-filesystem==0.31.0
termcolor==3.1.0
tf_keras==2.19.0
torch==2.8.0
torchvision==0.23.0
tqdm==4.67.1
typing-inspection==0.4.1
typing_extensions==4.14.1
tzdata==2025.2
ultralytics==8.3.198
ultralytics-thop==2.0.17
urllib3==2.5.0
Werkzeug==3.1.3
wrapt==1.17.3

### Reproducible example

```Python
async def search_face(
        top_k: int = Form(1, description="Döndürülecek en benzer sonuç sayısı."),
        collection_name: str = Form(DEFAULT_SEARCH_COLLECTION, description="Aramanın yapılacağı koleksiyon adı."),
        file: UploadFile = File(...)
):
    image_bytes = await file.read()
    query_vector = get_single_embedding_for_search(image_bytes)
    if query_vector is None:
        raise HTTPException(status_code=400, detail="Yüklenen resimde geçerli bir yüz tespit edilemedi.")

    hits = client.search(collection_name=collection_name, query_vector=query_vector, limit=top_k)

    if not hits: return {"status": "not_found", "results": []}
    results = [{"qdrant_id": hit.id, "score": hit.score, "payload": hit.payload} for hit in hits]

    # Sonuçları JSON uyumlu hale getir
    results = convert_numpy_types(results)

    return {"status": "success", "results": results}
```

### Additional Info

I am using Deepface with Qdrant, and I have indexed nearly 30 million face vectors into Qdrant using Deepface. I am using the ArcFace model and the RetinaFace detector backend. In my tests, I have achieved largely accurate results, and at this point, I want to extend my thanks to the developer team.

Normally, when I perform a search, one or two faces are found in the 0.7-1.0 similarity range. However, in some of my searches, the system detects 15-20 irrelevant individuals with a very high similarity range of 0.9-0.99.

I suspect this is due to corrupt or faulty vector extractions. How can I prevent this situation?
